lib/matplotlib/tests/test_contour.py

```diff
@@ -693,3 +693,25 @@ def test_contour_remove():
     assert ax.get_children() != orig_children
     cs.remove()
     assert ax.get_children() == orig_children
+
+
+def test_contour_z_kwargonly():
+    # Smoke test for GH#24743
+    # Passing only kwargs
+    plt.contour(Z=np.random.rand(30, 30))
+
+
+def test_contour_xyz_kwargsonly():
+    plt.contour(
```

Comment:
```suggestion
    plt.contour(Z=np.random.rand(30, 30))
    plt.contour(
```
although the test naming is not as good anymore.

Alternative: add a comment about smoke test for kwargs for `test_contour_xyz_kwargsonly` so that it is clear why the test is there.

Reply:
I feel that since those are 2 separate cases (and hence different execution paths), it is worth having different tests for each. Just so that we can pinpoint exactly which part fails in a test. Will change it if that doesn't go well with pre-established conventions.

Reply:
pytest prints out the traceback; no need to make a separate test unless there's some large difference between them.

Reply:
Okay. I'll change it to one test then.